src/mgmt/cli_mgmt.c

```diff
@@ -243,9 +245,18 @@ get_attach_info(const char *name, bool all_ranks, struct dc_mgmt_sys_info *info,
 		D_GOTO(out, rc);
 	}
 
+	ofi_interface = getenv("OFI_INTERFACE");
+	if (ofi_interface)
+		D_INFO("Using client provided OFI_INTERFACE: %s\n", ofi_interface);
+	ofi_domain = getenv("OFI_DOMAIN");
+	if (ofi_domain)
+		D_INFO("Using client provided OFI_DOMAIN: %s\n", ofi_domain);
```

Comment:
This change is to tell the agent about the client's desired network details, if they were set in the environment by the application user, right? What happens if the user sets some crazy stuff (e.g. copy/paste from some other env without understanding what they're for)?

I assume that there will be some logging on the agent side, but wondering how the error(s) will manifest to the application user.

Reply:
The primary goal of these changes is to choose the correct provider URIs/hint based on the client's env variables. I.e. pick a secondary provider if that's all the requested iface/domain enables. Initially I was planning to do more to validate the iface/domain, but after a couple conversations last week, I came to the conclusion that we shouldn't validate, and instead let them blindly override like before.

Basically I'm thinking of cases where we're adding new providers that aren't detected by the agent yet, like the recent development on UCX support. Developers are the only ones who should really be using these env variables.

Reply:
Hmm. I'm a little skeptical about the assertion that only developers will set the variables. Users are known for doing strange and unexpected things (whether they mean to or not). I was thinking that the agent config could be used to override the autoconfig behavior, but maybe I'm not remembering some details.

Reply:
Added some logging like we discussed, and I ended up moving things around a little. The errors printed to the agent log should at least provide a hint to folks doing something they're not supposed to.